src/parsec.rs

```diff
@@ -89,11 +133,19 @@ impl ParsecMap {
         pub_id: id::PublicId,
         log_ident: &LogIdent,
     ) -> bool {
-        let parsec = if let Some(parsec) = self.map.get_mut(&msg_version) {
-            parsec
-        } else {
-            return false;
-        };
+        let (parsec, size_counter) =
+            if let Some((parsec, size_counter)) = self.map.get_mut(&msg_version) {
+                (parsec, size_counter)
+            } else {
+                return false;
+            };
+
+        size_counter.increase_size(serialisation::serialised_size(&response));
```

Comment:
A struct for (parsec, size_counter) would probably allow you to avoid duplication here.

Reply:
I've wrapped those two in a struct in a fixup commit